### PR TITLE
Add compatibility to Python 3 (and retain Python 2)

### PR DESCRIPTION
**Pull request by mlell**
This PR tries to make redo also available for Python 3. I came across this as I make data analyses 
using redo and Python 3 and using two Pythons struck me as odd.

I tested this version under Python 2.7  and Python 3.7 and all tests pass.

### CODE REVIEW

**Comment by iMichka** [comment]
Hi. Homebrew maintainer here. Do you think this will make any progress before the Python2 EOL end of this year? We may need to remove this package from our package manager at one point or another: https://github.com/Homebrew/homebrew-core/issues/47050

**Comment by burdiyan** [comment]
I ran into this issue when I was trying to install redo with brew as per docs, and it couldn't find the package :(

@apenwarr do you think this PR could be merged?

**Comment by ACleverDisguise** [comment]
> @apenwarr do you think this PR could be merged?

Given that nothing seems to have touched the repo in almost a year, I'd say the answer to that is clearly "no".

Probably time to fork.

**Comment by burdiyan** [comment]
This program is used heavily at Talescale according to Twitter. So forking would create unnecessary fragmentation and would be a bit too radical. Probably this particular issue is not a problem for Talescale, and there’s very little time for maintenance.

But given the fact that redo was removed from Homebrew this issue seems to be pretty valuable for the community.

**Comment by burdiyan** [comment]
Just for the context https://twitter.com/apenwarr/status/1232922279983624192?s=21

**Comment by ACleverDisguise** [comment]
So, according to the maker of this, it is literally better to have a piece of software that's unusable than to have one that's a bit slow until he gets around to rewriting it in another language entirely.  (I nominate Rexx!)

Forking sounds a whole lot better now, doesn't it?

Or, in my case, especially given that redo doesn't support one of my common use cases well (multiple-inputs to multiple outputs), just not using redo at all sounds about right.

**Comment by apenwarr** [comment]
For the record, any distro that drops python2 just because it's "end of life" is actively user hostile. There are lots of unmaintained upstream packages that distros continue to include.

But anyway, I do appreciate the work on this pull request, especially that you took the time to not break python2 support. Merged. Thanks!

File: docs/md2man.py [hunk 1]
```diff
@@ -1,5 +1,11 @@
+from __future__ import print_function
 import sys, os, markdown, re
-from BeautifulSoup import BeautifulSoup
+try:
+    from BeautifulSoup import BeautifulSoup
+    bsver = 3
+except ModuleNotFoundError:
```
Comment (magical) [inline docs/md2man.py]: I don't think ModuleNotFoundError exists in python 2.

**Comment by apenwarr** [comment]
This will only trigger an error if BeautifulSoup is not available, and we
were going to get an error in that case anyway. So it's not ideal but it's
also not too terrible I guess.

**Comment by timmc** [comment]
Adding a note for anyone else who had previously built this version and later found it not working: If you uninstall python2, your built copy of redo may still be referencing `/usr/bin/python`, which will no longer exist (at least on Debian). Just rebuild and your redo scripts will start referencing `/usr/bin/python3` instead.

**Pull request merged**